validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -488,19 +489,88 @@ def get_batch_execution_result(self, batch_signature):
                                           clean_up=True)
             return BatchExecutionResult(is_valid=True, state_hash=state_hash)
 
+    def _is_predecessor_of_possible_successor(self,
+                                              txn_id,
+                                              possible_successor):
+        return txn_id in self._txn_predecessors[possible_successor]
+
+    def _txn_has_result(self, txn_id):
+        return txn_id in self._txn_results
+
+    def _is_in_same_batch(self, txn_id_1, txn_id_2):
+        return self._batches_by_txn_id[txn_id_1] == \
+               self._batches_by_txn_id[txn_id_2]
+
+    def _is_txn_to_replay(self, txn_id, possible_successor, already_seen):
+        """Decide if possible_successor should be replayed.
+
+        Args:
+            txn_id (str): Id of txn in failed batch.
+            possible_successor (str): Id of txn to possibly replay.
+            already_seen (list): A list of possible_successors that have
+                been replayed.
+
+        Returns:
+            (bool): If the possible_successor should be replayed.
+        """
+
+        is_successor = self._is_predecessor_of_possible_successor(
+            txn_id,
+            possible_successor)
+        in_different_batch = not self._is_in_same_batch(txn_id,
+                                                        possible_successor)
+        has_not_been_seen = possible_successor not in already_seen
+
+        return is_successor and in_different_batch and has_not_been_seen
+
+    def _remove_subsequent_result_because_of_batch_failure(self, sig):
+        """Remove transactions from scheduled and txn_results for
+         successors of txns in a batch that failed for another transaction.
```

Comment:
I suggest rewording this. It is unclear what you mean by "for another transaction". I also suggest adding that they will be rescheduled instead of just "This gives the replay ability."